pipeline/settings.py

```diff
@@ -28,3 +29,7 @@
 REDUCED_DATA_DIR = ROOT / "reduced"
 MODEL_FILE = ROOT.parent / "model.json"
 MODEL_SERVER_FILE = ROOT.parent / "serve_model.py"
+
+lock_generate = FileLock("generate.lock")
+lock_json_to_parquet = FileLock("json_to_parquet.lock")
+lock_compact = FileLock("compact.lock")
```

Comment:
I guess these work even in the cloud setting because this is all on the Prefect VM, right?  But they have to be file locks rather than `threading.Lock`s because prefect will create many processes?

Reply:
That's correct

Also, I was using Prefect's built in concurrency limiting functionality https://docs.prefect.io/latest/guides/global-concurrency-limits/ but encountered, which I like the idea of more than using filelocks. But we would sporadically run into deadlocks which is why I switched back to filelocks.

Also opened this issue https://github.com/PrefectHQ/prefect/issues/12015 as a possible alternative

Reply:
Probably what we actually want here is something like ...

```python
if lock.acquired:
    return

with lock:
    ...
```

Because we don't want things to pile up.

Not saying that we should actually do this, just mentioning it.  Short term we could also make our own limiting decorator.  My guess is that this makes more sense / is alot easier with `prefect serve` than it does with other Prefect deployment strategies.